### PR TITLE
[GLUTEN-4073][VL] Warn users about disabling of tests and storage adapters when ENABLE_BENCHMARK=ON

### PR DESCRIPTION
**Pull request by surnaik**
## What changes were proposed in this pull request?

Disables Tests and Connectors when ENABLE_BENCHMARK=ON and clearly echoes the message to the user.

Fixes: #4073 

## How was this patch tested?

Manual testing

### CODE REVIEW

**Comment by surnaik** [comment]
@FelixYBW @zhouyuan Please review. Thanks!

**Comment by surnaik** [comment]
Looking back into this, I think this might be redundant. Since the filesystem is registered during HiveConnector::initialize(), the error seems to be coming because the BUILD_BENCHMARKS=ON was set during the build of velox, which disabled VELOX_ENABLE_GCS. Might need a fix in the build script

**Comment by surnaik** [comment]
I had ENABLE_BENCHMARK=ON, this makes Velox disable tests and connectors. I have now echoed a clear message to the user if ENABLE_BENCHMARK=ON

**Comment by surnaik** [comment]
Fixed Tests

**Comment by surnaik** [comment]
Failure seems transient - 
503 Service Unavailable ProxyInfo{host='child-prc.intel.com', userName='null', port=913, type='https', nonProxyHosts='null'}

and another one is unable to download lzo
error: Failed to download from mirror set
error: http://www.oberhumer.com/opensource/lzo/download/lzo-2.10.tar.gz: curl failed to download with exit code 22
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

curl: (22) The requested URL returned error: 503 Service Unavailable

**Comment by zhouyuan** [comment]
@surnaik thanks for the help, the change looks good to me, do you happen to know why Velox needs to disable tests and connectors if benchmark is enabled?
https://github.com/facebookincubator/velox/blob/aaaa0791008267d8be0a5c141db6da261aba8892/CMakeLists.txt#L160

thanks, -yuan

**Comment by surnaik** [comment]
This was done to speed up compilation according the PR - https://github.com/facebookincubator/velox/commit/bcccc2a52e562716ba25d0c506aaf8217f101e6e

**Comment by surnaik** [comment]
@zhouyuan Seems like faster compile time is the only reason for disabling testing and connectors.

**Review by zhouyuan — APPROVED** [review]
👍

File: ep/build-velox/src/build_velox.sh [hunk 1]
```diff
@@ -96,23 +96,26 @@ function compile {
...
-    COMPILE_OPTION="$COMPILE_OPTION -DVELOX_ENABLE_ABFS=ON"
-  fi
   if [ $ENABLE_S3 == "ON" ]; then
     COMPILE_OPTION="$COMPILE_OPTION -DVELOX_ENABLE_S3=ON"
   fi
-  if [ $ENABLE_GCS == "ON" ]; then
-    COMPILE_OPTION="$COMPILE_OPTION -DVELOX_ENABLE_GCS=ON"
+  # If ENABLE_BENCHMARK == ON, Velox disables tests and connectors
+  if [ $ENABLE_BENCHMARK == "OFF" ]; then
+    if [ $ENABLE_TESTS == "ON" ]; then
```
Comment (dcoliversun) [inline ep/build-velox/src/build_velox.sh:107]: This seems to be a regression of https://github.com/oap-project/gluten/pull/4176. Script `./dev/package.sh` is broken. cc @zhouyuan @surnaik

Comment (surnaik) [reply ep/build-velox/src/build_velox.sh:107]: This is due to a recent commit in Velox - 

https://github.com/facebookincubator/velox/commit/f2cdf813be50cc165a104b93cabbb56319bd4822

**Pull request merged**